### DISCUSSION TITLE
IMU Integration with ROS

### DISCUSSION THREAD

**Discussion by anvar-mahammad** [OpenVSLAM-Community/openvslam_ros#74] [Q&A] [answered]
Hi! 

I installed the OpenVSLAM branch with IMU integration and tried to install ROS Package as well. I am getting an error saying openvslam::system doesn't have a member called relocalize_by_pose_2d when building openvslam_ros.cc. I was wondering when you guys will be able to have ROS package that also has the IMU Integration? or is there a fix to this except just manually adding the classes to the system.cc and tracking_module.cc, which I suspect will kill the whole purpose of having IMU integration.

Best regards,
Anvar

**Comment by ymd-stella** [comment, marked as answer]
I've rebased it. I think it works now. By the way, do you know that IMU integration is not yet complete?

**Reply by anvar-mahammad** [reply]
Thanks for the heads-up! Do you have a timeline on when it will be ready? I have been following the original and now the community openvslam and it has been in the to do list for quite a while. Me and my friend are trying to build a home robot so it is a steep learning curve!  Also I hade another question. I am struggling with camera calibration. I used Ros Melodic Calibration node and I get an error saying:

yaml-cpp error at line 0 column 0: invalid node; this may result from using a map iterator as a sequence iterator or vice versa

What do you use to obtain calibration data for your camera?

I have also tried to use one of the example config files to no avail. My calibration file contents are the following:

image_width: 820
image_height: 616
camera_name: csi_cam_0
camera_matrix:
  rows: 3
  cols: 3
  data: [ 576.34139,    0.     ,  429.69978,
            0.     ,  771.38529,  310.11047,
            0.     ,    0.     ,    1.     ]
camera_model: plumb_bob
distortion_coefficients:
  rows: 1
  cols: 5
  data: [-0.954283, 0.787176, 0.018706, -0.027956, 0.000000]
rectification_matrix:
  rows: 3
  cols: 3
  data: [ 1.,  0.,  0.,
          0.,  1.,  0.,
          0.,  0.,  1.]
projection_matrix:
  rows: 3
  cols: 4
  data: [ 424.38644,    0.     ,  404.98434,    0.     ,
            0.     ,  627.6886 ,  326.51277,    0.     ,
            0.     ,    0.     ,    1.     ,    0.     ]

**Reply by ymd-stella** [reply]
Please create another discussion if you need, as I will check it later.

**Reply by ymd-stella** [reply]
For a short answer, read the following.

https://github.com/OpenVSLAM-Community/openvslam/blob/main/example/euroc/EuRoC_stereo.yaml
https://openvslam-community.readthedocs.io/en/latest/simple_tutorial.html
https://openvslam-community.readthedocs.io/en/latest/ros_package.html#examples
